src/components/PizzaBlock.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useMemo, useState } from 'react';
+import { addItem } from '../redux/slices/cartSlice';
+import { useAppDispatch, useAppSelector } from '../redux/store';
+import { PitsaType } from '../types/pitsaType';
+import { PlusSVG } from '../assets/svgs/PlusSVG';
+
+const typeNames = ['тонкое', 'традиционное'];
+
+function PizzaBlock({
+  id,
+  title,
+  price,
+  imageUrl,
+  sizes,
+  types,
+}: PitsaType) {
+  const dispatch = useAppDispatch();
+  const cartItems = useAppSelector((state) =>
+    state.cartSlice.items.filter((obj) => obj.id === id)
+  );
+  const [activeType, setActiveType] = useState(0);
+  const [activeSize, setActiveSize] = useState(0);
+
+  const addedCount = useMemo(() => {
+    let totalCount = 0;
+    cartItems.forEach((pitsa) => {
+      totalCount += pitsa.count;
+    });
+    return totalCount;
+  }, [cartItems]);
+
+  const onClickAdd = () => {
+    const item = {
+      id,
+      title,
+      price,
+      imageUrl,
+      type: typeNames[activeType],
+      size: sizes[activeSize],
+    };
+    dispatch(addItem(item));
```

Comment:
This constant looks redundant use object inside action argument
```suggestion
    dispatch(addItem({
      id,
      title,
      price,
      imageUrl,
      type: typeNames[activeType],
      size: sizes[activeSize],
    }));
```